spring-cloud-gcp-data-firestore/src/test/java/com/google/cloud/spring/data/firestore/it/FirestoreIntegrationTests.java

```diff
@@ -380,4 +262,27 @@ void deleteTest() {
     this.firestoreTemplate.deleteAll(User.class).block();
     assertThat(this.firestoreTemplate.count(User.class).block()).isZero();
   }
+
+  @Test
+  void transactionTest() {
+    User alice = new User("Alice", 29);
+    User bob = new User("Bob", 60);
+
+    User user = new User(null, 40);
+
+    DefaultTransactionDefinition transactionDefinition = new DefaultTransactionDefinition();
+    transactionDefinition.setReadOnly(false);
+    TransactionalOperator operator =
+        TransactionalOperator.create(this.txManager, transactionDefinition);
+
+    this.firestoreTemplate
+        .save(alice)
+        .then(this.firestoreTemplate.save(bob))
+        .then(this.firestoreTemplate.save(user))
+        .as(operator::transactional)
+        .block();
+
+    assertThat(this.firestoreTemplate.findAll(User.class).collectList().block())
+        .containsExactlyInAnyOrder(bob, alice, user);
```

Comment:
I know you're just relocating this test and not touching the logic, but this exact-order assertion seems odd...